mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -341,6 +323,48 @@ object CaseKeywordCompletion:
         defnSymbols.getOrElse(semancticName, -1)
       }
 
+  def subclassesForType(tpe: Type)(using Context): List[Symbol] =
+    // Symbol must be a subclass of every `AndType`
+    def getAndTypes(tpe: Type, acc: List[Type]): List[Type] =
+      tpe match
+        case AndType(tp1, tp2) =>
+          getAndTypes(tp2, getAndTypes(tp1, acc))
+        case t => t :: acc
+
+    val andTypes = getAndTypes(tpe, List.empty)
+
+    def getParentTypes(tpe: Type, acc: List[Symbol]): List[Symbol] =
+      tpe match
+        case AndType(tp1, tp2) =>
+          getParentTypes(tp2, getParentTypes(tp1, acc))
+        case OrType(tp1, tp2) =>
```

Comment:
OrType is no neccessarily a parent, is it? It's one of, we might need separate paths for OR?

Reply:
I'm not sure what you mean. I added some scaladoc according to @kpodsiad suggestion, maybe it's clearer now